rustls/src/crypto/aws_lc_rs/ticketer.rs

```diff
@@ -153,48 +185,52 @@ impl ProducesTickets for AeadTicketer {
-        // tickets for a different ticketer (see `TicketSwitcher`), while including it
-        // in the AAD ensures it is authenticated independent of that check and that
-        // any attempted attack on the integrity such as [^1] must happen for each
-        // `key_label`, not over a population of potential keys.  this approach
-        // is overall similar to [^2].
-        //
-        // [^1]: https://eprint.iacr.org/2020/1491.pdf
-        // [^2]: "Authenticated Encryption with Key Identification", fig 6
-        //       <https://eprint.iacr.org/2022/1680.pdf>
-        if ConstantTimeEq::ct_ne(&self.key_name[..], alleged_key_name).into() {
-            #[cfg(debug_assertions)]
-            debug!("rejected ticket with wrong ticket_name");
-            return None;
-        }
-
-        // This won't fail since `nonce` has the required length.
-        let nonce = aead::Nonce::try_assume_unique_for_key(nonce).ok()?;
-
-        let mut out = Vec::from(ciphertext);
-
-        let plain_len = self
-            .key
-            .open_in_place(nonce, aead::Aad::from(alleged_key_name), &mut out)
-            .ok()?
-            .len();
-        out.truncate(plain_len);
+        // Split off the IV from the remaining ciphertext.
+        let (iv, ciphertext) = try_split_at(ciphertext, cipher::AES_CBC_IV_LEN)?;
+
+        // And finally, split the encrypted state from the tag.
+        let tag_len = self
+            .hmac_key
+            .algorithm()
+            .digest_algorithm()
+            .output_len();
+        let (enc_state, mac) = try_split_at(ciphertext, ciphertext.len() - tag_len)?;
+
+        // Reconstitute the HMAC data to verify the tag.
+        let mut hmac_data =
+            Vec::with_capacity(alleged_key_name.len() + iv.len() + 2 + enc_state.len());
+        hmac_data.extend(alleged_key_name);
+        hmac_data.extend(iv);
+        hmac_data.extend(
+            u16::try_from(enc_state.len())
+                .ok()?
+                .to_be_bytes(),
+        );
+        hmac_data.extend(enc_state);
+        hmac::verify(&self.hmac_key, &hmac_data, mac).ok()?;
+
+        // Convert the raw IV back into an appropriate decryption context.
+        let iv = iv::FixedLength::try_from(iv).ok()?;
+        let dec_context = cipher::DecryptionContext::Iv128(iv);
+
+        // And finally, decrypt the encrypted state.
+        let mut out = Vec::from(enc_state);
+        let plaintext = self
+            .aes_decrypt_key
+            .decrypt(&mut out, dec_context)
+            .ok()?;
 
-        Some(out)
+        Some(plaintext.into())
     }
 }
 
-impl Debug for AeadTicketer {
+impl Debug for Rfc5077 {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        // Note: we deliberately omit the key from the debug output.
-        f.debug_struct("AeadTicketer")
-            .field("alg", &self.alg)
+        // Note: we deliberately omit keys from the debug output.
+        f.debug_struct("Rfc5077Ticketer")
```

Comment:
The type name is currently inconsistent with the code.

Reply:
This fell away with the return to using `Rfc5077Ticketer` as the name.